### PR TITLE
AV-182960 : Fix to support mixed protocols for ports in LoadBalancer services and Gateway Services API

### PR DESCRIPTION
**Pull request by DixitAakash**
This PR adds support for mixed protocols for ports in LoadBalancer services and Gateway Services API. This support was missing due to error in implementation.

### CODE REVIEW

**Comment by DixitAakash** [comment]
build ako

**Review by aaha97 — DISMISSED** [review]
LGTM

File: internal/nodes/avi_model_l4_translator.go [hunk 1]
```diff
@@ -100,21 +102,20 @@ func (o *AviObjectGraph) ConstructAviL4VsNode(svcObj *corev1.Service, key string
...
-
 		if license == lib.LicenseTypeEnterprise {
 			avi_vs_meta.NetworkProfile = utils.DEFAULT_TCP_NW_PROFILE
 		} else {
 			avi_vs_meta.NetworkProfile = utils.TCP_NW_FAST_PATH
 		}
+	} else if isUDP && !isTCP && !isSCTP {
+		avi_vs_meta.NetworkProfile = utils.SYSTEM_UDP_FAST_PATH
+	} else {
+		avi_vs_meta.NetworkProfile = utils.MIXED_NET_PROFILE
```
Comment (swathinsankaran) [inline internal/nodes/avi_model_l4_translator.go:118]: This logic seems to be repeated in three places. Is it possible to move it into a single place?

**Comment by swathinsankaran** [comment]
overall changes look good to me.

**Review by swathinsankaran — APPROVED** [review]

File: internal/nodes/avi_model_l4_translator.go (hunk 1 above)
Comment (DixitAakash) [reply internal/nodes/avi_model_l4_translator.go]: Have moved it to a new function.

**Comment by DixitAakash** [comment]
@aaha97 @swathinsankaran since you both had already approved the PR, I would like to let you know that in the subsequent commits I am just setting the network profile to either TCP proxy or fast path based on the license. Previously, it was mostly fast path.

**Comment by DixitAakash** [comment]
build ako

File: internal/nodes/avi_model_l4_translator.go [hunk 2]
```diff
@@ -662,3 +650,23 @@ func buildPoolWithL4Rule(key string, pool *AviPoolNode, l4Rule *akov1alpha2.L4Ru
...
+
+// In case the VS has services that are a mix of TCP and UDP/SCTP sockets,
+// we create the VS with global network profile TCP Proxy or Fast Path based on license,
+// and override required services with UDP Fast Path or SCTP proxy. Having a separate
+// internally used network profile (MIXED_NET_PROFILE) helps ensure PUT calls
+// on existing VSes.
+func getNetworkProfile(isSCTP, isTCP, isUDP bool) string {
+	if isSCTP && !isTCP && !isUDP {
+		return utils.SYSTEM_SCTP_PROXY
+	} else if isTCP && !isUDP && !isSCTP {
```
Comment (DixitAakash) [inline internal/nodes/avi_model_l4_translator.go:662]: @abhishekbsingh As discussed, can you please check this scenario for advanced l4. The network profile was previously set to TCP fast path here but now being set based on license.

**Review by akshayhavile — APPROVED** [review]

File: internal/nodes/avi_model_l4_translator.go (hunk 2 above)
Comment (abhishekbsingh) [reply internal/nodes/avi_model_l4_translator.go:662]: I think it should be okay, nit: else is not required as each if block has a return in it

**Review by akshayhavile — APPROVED** [review]

**Comment by DixitAakash** [comment]
build ako

File: internal/nodes/avi_model_l4_translator.go (hunk 2 above)
Comment (DixitAakash) [reply internal/nodes/avi_model_l4_translator.go]: Have removed the else statements.

**Review by swathinsankaran — APPROVED** [review]

**Comment by DixitAakash** [comment]
build ako

**Comment by DixitAakash** [comment]
build ako

**Pull request merged**